filebeat/input/event.go

```diff
@@ -105,6 +105,10 @@ func mergeJSONFields(e *Event, event common.MapStr) {
 			}
 		}
 	} else {
-		event["json"] = e.JSONFields
+		key := e.JSONConfig.JsonKey
+		if len(key) == 0 {
+			key = "json"
```

Comment:
It would be nice if this check could be done once at initialization and then set the value to the default if it's empty. The usual pattern is to initialize the `JsonKey` value to its default value prior to unpacking the user's config; maybe that can be done in this case.